codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Codec.java

```diff
@@ -0,0 +1,107 @@
+ * this API. This API is targeted to eventually replace or reduce the need for the
+ * Http2Connection-based API.
+ *
+ * <p>This handler notifies the pipeline of channel events, such as {@link Http2GoAwayFrame}. It
+ * is also capable of writing such messages. Directly writing {@link Http2StreamFrame}s for this
+ * handler is unsupported.
+ *
+ * <h3>Child Channels</h3>
+ *
+ * <p>When a new stream is created, a new {@link Channel} is created for it. Applications send and
+ * receive {@link Http2StreamFrame}s on the created channel. The {@link Http2StreamFrame#stream} is
+ * expected to be {@code null}, but the channel can use the field for its own bookkeeping. {@link
+ * ByteBuf}s cannot be processed by the channel; all writes that reach the head of the pipeline must
+ * be an instance of {@link Http2StreamFrame}.  Writes that reach the head of the pipeline are
+ * processed directly by this handler and cannot be intercepted.
+ *
+ * <p>The child channel will be notified of user events that impact the stream, such as {@link
+ * Http2GoAwayFrame} and {@link Http2ResetFrame}, as soon as they occur. Although {@code
+ * Http2GoAwayFrame} and {@code Http2ResetFrame} signify that the remote is ignoring further
+ * communication, closing of the channel is delayed until any inbound queue is drained with {@link
+ * Channel#read()}, which follows the default behavior of channels in Netty. Applications are
+ * free to close the channel in response to such events if they don't have use for any queued
+ * messages.
+ *
+ * <p>{@link ChannelConfig#setMaxMessagesPerRead(int)} and {@link
+ * ChannelConfig#setAutoRead(boolean)} are supported.
+ */
+@UnstableApi
+public final class Http2Codec extends ChannelDuplexHandler {
+
+    private final Http2FrameCodec frameCodec;
+    private final Http2MultiplexCodec multiplexCodec;
+
+    /**
+     * Construct a new handler whose child channels run in the same event loop as this handler.
+     *
+     * @param server {@code true} this is a server
+     * @param streamHandler the handler added to channels for remotely-created streams. It must be
+     *     {@link ChannelHandler.Sharable}.
+     */
+    public Http2Codec(boolean server, ChannelHandler streamHandler) {
+        this(server, streamHandler, null);
+    }
+
+    /**
+     * Construct a new handler whose child channels run in a different event loop.
+     *
+     * @param server {@code true} this is a server
+     * @param streamHandler the handler added to channels for remotely-created streams. It must be
+     *     {@link ChannelHandler.Sharable}.
+     * @param streamGroup event loop for registering child channels
+     */
+    public Http2Codec(boolean server, ChannelHandler streamHandler,
+                      EventLoopGroup streamGroup) {
+        this(server, streamHandler, streamGroup, new DefaultHttp2FrameWriter());
+    }
+
+    Http2Codec(boolean server, ChannelHandler streamHandler,
```

Comment:
Yes

> Am 30.05.2016 um 16:15 schrieb Jakob Buchgraber <notifications@github.com>:
> 
> In codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Codec.java:
> 
> > +    }
> > +
> > +    /**
> > +     * Construct a new handler whose child channels run in a different event loop.
> > +     *
> > +     * @param server {@code true} this is a server
> > +     * @param streamHandler the handler added to channels for remotely-created streams. It must be
> > +     *     {@link ChannelHandler.Sharable}.
> > +     * @param streamGroup event loop for registering child channels
> > +     */
> > +    public Http2Codec(boolean server, ChannelHandler streamHandler,
> > +                      EventLoopGroup streamGroup) {
> > +        this(server, streamHandler, streamGroup, new DefaultHttp2FrameWriter());
> > +    }
> > +
> > +    Http2Codec(boolean server, ChannelHandler streamHandler,
> @normanmaurer once I ll add tests to Http2CodecTest 😅, this constructor will be needed there. Should I add a comment // Visible for testing?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>